src/BcPrebidVast.js

```diff
@@ -636,11 +641,76 @@ function loadMolPlugin(callback) {
     }
 }
 
+function getAdRendererFromAdOptions(adOptions) {
+	// if adRenderer option is present use it for all ads
+	if (adOptions.hasOwnProperty('adRenderer') && adOptions.adRenderer &&
+		(adOptions.adRenderer === _rendererNames.MOL || adOptions.adRenderer === _rendererNames.IMA)) {
+		return {adRenderer: adOptions.adRenderer, needBreak: true};
+	}
+	// for DFP use IMA plugin as renderer
+	if (adOptions.hasOwnProperty('dfpParameters')) {
+		return {adRenderer: _rendererNames.IMA, needBreak: false};
+	}
+	return null;
+}
+
+function setAdRenderer(options) {
+	if (options) {
+		var adRenderer = _rendererNames.MOL;
+		var rendObj;
+		var i;
+		if (Array.isArray(options)) {
+			for (i = 0; i < options.length; i++) {
+				rendObj = getAdRendererFromAdOptions(options[i]);
+				if (rendObj) {
+					adRenderer = rendObj.adRenderer;
+					if (rendObj.needBreak) {
+						break;
+					}
+				}
+			}
+			if (adRenderer) {
+				for (i = 0; i < options.length; i++) {
+					options[i].adRenderer = adRenderer;
+				}
+			}
+		}
+		else {
+			// array in brightcove studio converted to object {0: {...}, 1: {...}, ...}
+			if (options.hasOwnProperty('0')) {
```

Comment:
This second loop is exactly the same as the one above. Can we avoid repetition (keep it DRY)? Maybe this can be moved into a 2nd helper method that you can pass either an Array or the Array-like Brightcove object to.

Reply:
Made code optimization.